drivers/led/is31fl3216a.c

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023 Endor AG
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT issi_is31fl3216a
+
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/led.h>
+#include <zephyr/logging/log.h>
+
+#define IS31FL3216A_REG_CONFIG		0x00
+#define IS31FL3216A_REG_CTL_1		0x01
+#define IS31FL3216A_REG_CTL_2		0x02
+#define IS31FL3216A_REG_LIGHT_EFFECT	0x03
+#define IS31FL3216A_REG_CHANNEL_CONFIG	0x04
+#define IS31FL3216A_REG_PWM_FIRST	0x10
+#define IS31FL3216A_REG_PWM_LAST	0x1F
+#define IS31FL3216A_REG_UPDATE		0xB0
+
+#define IS31FL3216A_MAX_LEDS		16
+
+LOG_MODULE_REGISTER(is31fl3216a, CONFIG_LED_LOG_LEVEL);
+
+struct is31fl3216a_cfg {
+	struct i2c_dt_spec i2c;
+};
+
+static int is31fl3216a_write_reg(const struct i2c_dt_spec *i2c, uint8_t reg,
+				 uint8_t val)
```

Comment:
Suggestion: since this controllers supports auto increment of I2C addresses then you could use that to write adjacent registers with a single `i2c_write_dt` call.

Reply:
Will do with  ``write_channels`` implementation.

*N.B*: The second call in ``is31fl3216a_led_set_brightness`` is necessary to request the chip to load the new PWM value because it is not be adjacent to the PWM register.

> B0h Update Register
>
> The data sent to the PWM Register will be stored in temporary registers. A write operation of “0000 0000” to the Update Register is required to update the registers.